wporg-two-factor.php

```diff
@@ -192,16 +192,18 @@ function remove_super_admins_until_2fa_enabled() : void {
  * That is necessary even though we'll redirect all requests to their profile, because otherwise they could still
  * perform privileged actions on the front end, via the REST API, etc.
  */
-function remove_capabilities_until_2fa_enabled( array $allcaps, array $caps, array $args, WP_User $user ) : array {
-	if ( 0 === $user->ID || ! user_requires_2fa( $user ) ) {
+function remove_capabilities_until_2fa_enabled( array $allcaps, array $caps, array $args ) : array {
+	$current_user = wp_get_current_user();
+
+	if ( 0 === $current_user->ID || ! user_requires_2fa( $current_user ) ) {
 		return $allcaps;
 	}
```

Comment:
I think this should respect the passed $user, but the issue is that it shouldn't be performing any actions when the user having it's caps checked doesn't match the current logged in user.

Basically, revert all changes here and instead add a `$user->ID !== get_current_user_id()` to the early return.
```suggestion
function remove_capabilities_until_2fa_enabled( array $allcaps, array $caps, array $args, WP_User $user ) : array {
	if ( 0 === $user->ID || $user->ID !== get_current_user_id() || ! user_requires_2fa( $user ) ) {
		return $allcaps;
	}
```

Reply:
Thanks. I left this in the draft because I see that we use `wp_get_current_user` in other filters but not here. I wasn't familiar with when this filter is applied and was going to look into it and figure out why it was being called within the context of the forum tables. If it's as simple as comparing the current user then cool.